llama_toolchain/telemetry/api/api.py

```diff
@@ -6,170 +6,91 @@
 
 from datetime import datetime
 from enum import Enum
-
-from typing import Any, Dict, List, Optional, Protocol, Union
+from typing import Any, Dict, Literal, Optional, Protocol, Union
 
 from llama_models.schema_utils import json_schema_type, webmethod
-
-from pydantic import BaseModel
+from pydantic import BaseModel, Field
+from typing_extensions import Annotated
 
 
 @json_schema_type
-class ExperimentStatus(Enum):
-    NOT_STARTED = "not_started"
-    RUNNING = "running"
-    COMPLETED = "completed"
-    FAILED = "failed"
+class SpanStatus(Enum):
+    OK = "ok"
+    ERROR = "error"
 
 
 @json_schema_type
-class Experiment(BaseModel):
-    id: str
+class Span(BaseModel):
+    span_id: str
+    trace_id: str
+    parent_span_id: Optional[str] = None
     name: str
-    status: ExperimentStatus
-    created_at: datetime
-    updated_at: datetime
-    metadata: Dict[str, Any]
+    start_time: datetime
+    end_time: Optional[datetime] = None
+    attributes: Optional[Dict[str, Any]] = Field(default_factory=dict)
 
 
 @json_schema_type
-class Run(BaseModel):
-    id: str
-    experiment_id: str
-    status: str
-    started_at: datetime
-    ended_at: Optional[datetime]
-    metadata: Dict[str, Any]
+class Trace(BaseModel):
+    trace_id: str
+    root_span_id: str
+    start_time: datetime
+    end_time: Optional[datetime] = None
 
 
 @json_schema_type
-class Metric(BaseModel):
-    name: str
-    value: Union[float, int, str, bool]
-    timestamp: datetime
-    run_id: str
+class EventType(Enum):
+    LOG = "log"
+    SPAN_START = "span_start"
```

Comment:
will be good to have these types:
LOG,
STRUCTURED 
METRIC

at the top level.

Within STRUCTURED - we can add sub-event types like start end etc.

LOG sub-event types would be the severity enum

we could add other sub-event types for METRICS later